handlers/upload.go

```diff
@@ -64,6 +69,22 @@ func HasContentType(r *http.Request, mimetype string) bool {
 	return false
 }
 
+func (r UploadVODRequest) getSourceOutputURL() (*url.URL, error) {
+	for _, o := range r.OutputLocations {
+		if o.Outputs.SourceSegments {
+			return url.Parse(o.URL)
+		}
+	}
+	return nil, nil
+}
+
+func (r UploadVODRequest) getTargetURL() (*url.URL, error) {
+	if len(r.OutputLocations) < 1 {
+		return nil, errors2.New("no output locations")
+	}
+	return url.Parse(r.OutputLocations[0].URL)
```

Comment:
Should this be a similar logic looking for the `output` with `TranscodedSegments: true`?

Reply:
Yeah, I think you're right. I can change it that way.

Reply:
With the only difference is that we currently don't accept requests without `TranscodedSegments: true`, so it'll return an error.